### PR TITLE
Fix blocktransfer stall due to lost ACK

### PR DESCRIPTION
**Pull request by qleisan**
Signed-off-by: Leif Sandstrom <leif.sandstrom@husqvarnagroup.com>

Fix for #555

### CODE REVIEW

File: coap/block.c [hunk 1]
```diff
@@ -247,8 +247,8 @@ uint8_t prv_coap_block_handler(lwm2m_block_data_t ** pBlockDataHead,
         }
 
         if (blockNum <= blockData->blockNum){
-            // this is a retransmissiion, ignore
-            return COAP_IGNORE;
+            // this is a retransmission
+            return COAP_231_CONTINUE;
```
Comment (qleisan) [inline coap/block.c:251]: what if this is the last block...?

**Comment by qleisan** [comment]
Tested that the last ACK of the blocktransfer can be lost and will be resent as 2.04
Tested that an ACK in the middle of the blocktransfer can be lost and will be resent as 2.31

File: coap/block.c (hunk 1 above)
Comment (qleisan) [reply coap/block.c]: pushed new improved fix

**Comment by rettichschnidi** [comment]
Any chance to get this fix unit-tested?

**Comment by qleisan** [comment]
"Integration testing" is the only realistic way (as I see it) to get decent test coverage with realistic effort. And we should refactor some of the blocktransfer code once we have to make it cleaner and easier to understand.
I plan to look into integration testing now, happy to discuss best way forward

**Review by tuve — COMMENTED** [review]
This looks fine to me. One question related to this, do we ACK retransmissions if they are block transfers (as we also should)?

**Comment by sbernard31** [comment]
Shoul I integrate this one ?

**Comment by qleisan** [comment]
yes and yes!

**Comment by sbernard31** [comment]
Integrated in master (commit f4328b8f997ac7c8815f4f179695953cb193f5d3)

**Pull request closed without merging**